### PR TITLE
Shot vector property

### LINKED ISSUE #1164: Please expose device._shot_vector via public property
This is a features request: Please expose the shot vector via some public and documented property of the device. Thank you!

This is inspired by me needing to predict the output shape of certain QNodes and in case of a non trivial shot vector and all observables being owned by an `expval` the shape is 
```
(sum([shot_tuple.copies for shot_tuple in self.device._shot_vector]), len(self.measurements))
```
(where self is a tape).

### PR DESCRIPTION
**Pull request by Thenerdstation**
Fixes #1164

### CODE REVIEW

File: pennylane/_device.py [hunk 1]
```diff
@@ -264,6 +264,10 @@ def shots(self, shots):
                 "Shots must be a single non-negative integer or a sequence of non-negative integers."
             )
 
+    @property
+    def shot_vector(self):
```
Comment (josh146) [inline pennylane/_device.py:268]:
```suggestion
    def shot_vector(self):
    """list[.ShotTuple[int, int]]: Returns the shot vector, a sparse
    representation of the shot sequence used by the device
    when evaluating QNodes.

    **Example**

    >>> dev = qml.device("default.qubit", wires=2, shots=[3, 1, 2, 2, 2, 2, 6, 1, 1, 5, 12, 10, 10])
    >>> dev.shots
    57
    >>> dev.shot_vector
    [ShotTuple(shots=3, copies=1),
     ShotTuple(shots=1, copies=1),
     ShotTuple(shots=2, copies=4),
     ShotTuple(shots=6, copies=1),
     ShotTuple(shots=1, copies=2),
     ShotTuple(shots=5, copies=1),
     ShotTuple(shots=12, copies=1),
     ShotTuple(shots=10, copies=2)]

    The sparse representation of the shot
    sequence is returned, where tuples indicate the number of times a shot
    integer is repeated.
    """
```

Comment (josh146) [reply pennylane/_device.py:268]: I had this docstring lying around in one of my local files. I haven't tested it, but I recommend running the example code just to make sure that's what it outputs 😆

**Review by josh146 — CHANGES_REQUESTED** [review]
Thanks @Thenerdstation, just needs a docstring (have left one I had made earlier). Might also be worth adding this PR to the changelog. I don't think it needs its own entry, maybe just add the PR link to the existing shots refactor III

File: pennylane/_device.py [hunk 2]
```diff
@@ -264,6 +264,33 @@ def shots(self, shots):
...
+     ShotTuple(shots=6, copies=1),
+     ShotTuple(shots=1, copies=2),
+     ShotTuple(shots=5, copies=1),
+     ShotTuple(shots=12, copies=1),
+     ShotTuple(shots=10, copies=2)]
+
+    The sparse representation of the shot
+    sequence is returned, where tuples indicate the number of times a shot
+    integer is repeated.
+    """
```
Comment (josh146) [inline pennylane/_device.py:291]:
```suggestion
        """list[.ShotTuple[int, int]]: Returns the shot vector, a sparse
        representation of the shot sequence used by the device
        when evaluating QNodes.

        **Example**

        >>> dev = qml.device("default.qubit", wires=2, shots=[3, 1, 2, 2, 2, 2, 6, 1, 1, 5, 12, 10, 10])
        >>> dev.shots
        57
        >>> dev.shot_vector
        [ShotTuple(shots=3, copies=1),
         ShotTuple(shots=1, copies=1),
         ShotTuple(shots=2, copies=4),
         ShotTuple(shots=6, copies=1),
         ShotTuple(shots=1, copies=2),
         ShotTuple(shots=5, copies=1),
         ShotTuple(shots=12, copies=1),
         ShotTuple(shots=10, copies=2)]

        The sparse representation of the shot
        sequence is returned, where tuples indicate the number of times a shot
        integer is repeated.
        """
```

**Comment by Thenerdstation** [comment]
@josh146 I think you fixed your own change request so can we merge this?

**Review by josh146 — APPROVED** [review]

**Pull request merged**